File: hello-grpc/src/main/java/local/example/service/HelloService.java
package local.example.service;

import io.quarkus.grpc.GrpcService;
import io.smallrye.mutiny.Uni;

import local.example.hello.grpc.Greeter;
import local.example.hello.grpc.HelloReply;
import local.example.hello.grpc.HelloRequest;

@GrpcService
public class HelloService implements Greeter {

    @Override
    public Uni<HelloReply> sayHello(HelloRequest helloRequest) {
        return Uni.createFrom().item(
                () -> HelloReply.newBuilder().setMessage(
                        "Hello " + helloRequest.getName()
                ).build()
        );
    }
}

File: hello-service/src/main/kotlin/local/example/service/HelloService.kt
package local.example.service

import io.quarkus.grpc.GrpcService
import io.smallrye.mutiny.Uni

import local.example.gprc.Greeter
import local.example.gprc.HelloRequest
import local.example.grpc.HelloReply

@GrpcService
class HelloService : Greeter {
    override fun sayHello(request: HelloRequest): Uni<HelloReply> {
        return Uni.createFrom().item {
            HelloReply.newBuilder().setMessage("Hello " + request.name).build()
        }
    }
}